extensions/obviousAlexC/penPlus.js

```diff
@@ -154,18 +158,48 @@
     window.addEventListener("resize", updateCanvasSize);
     vm.runtime.on("STAGE_SIZE_CHANGED", () => {
       updateCanvasSize();
+      resizeCall = true;
     });
 
+    //Turbowarp
     vm.runtime.on("BEFORE_EXECUTE", () => {
-      if (
-        (renderer.useHighQualityRender
+      let calcSize = renderer.useHighQualityRender
+        ? [canvas.width, canvas.height]
+        : renderer._nativeSize;
+      if (calcSize[0] != nativeSize[0] || calcSize[1] != nativeSize[1]) {
+        nativeSize = renderer.useHighQualityRender
           ? [canvas.width, canvas.height]
-          : renderer._nativeSize) != nativeSize
-      ) {
+          : renderer._nativeSize;
+        updateCanvasSize();
+      }
+
+      if (resizeCall) {
         nativeSize = renderer.useHighQualityRender
           ? [canvas.width, canvas.height]
           : renderer._nativeSize;
         updateCanvasSize();
+        resizeCall = false;
+      }
+    });
+
+    //Penguinmod
+    vm.runtime.on("RUNTIME_STEP_START", () => {
```

Comment:
its on them to implement BEFORE_EXECUTE which would also fix other extensions.

Reply:
the rationale here is 

 - this is the TurboWarp extension gallery
 - we don't need to set a precedent of having every BEFORE_EXECUTE also have an identical RUNTIME_STEP_START
 - them fixing it will fix other extensions automatically
 - it takes them one line to implement

Reply:
I'm just trying to have it so I don't have to main multiple branches like I did for the old turboloader builds.

Reply:
hopefully other mods add support for APIs such as BEFORE_EXECUTE so you don't need to maintain separate versions like that

Reply:
They did